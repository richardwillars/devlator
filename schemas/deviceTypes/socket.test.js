const socketModule = require("./socket");
const constants = require("../../constants");
const eventsModule = require("../events");

const events = eventsModule(constants);

describe("models/socket", () => {
  it("should export the commands", () => {
    const socketModel = socketModule(events, constants);
    expect(socketModel.commands).toMatchSnapshot();
  });

  it("should export the events", () => {
    const socketModel = socketModule(events, constants);
    expect(socketModel.events).toMatchSnapshot();
  });
});
