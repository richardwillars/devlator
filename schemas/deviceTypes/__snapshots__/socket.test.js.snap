// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/socket should export the commands 1`] = `
Object {
  "off": Object {
    "description": "Turns the socket off",
    "friendly": "Turn off",
    "type": [Function],
  },
  "on": Object {
    "description": "Turns the socket on",
    "friendly": "Turn on",
    "type": [Function],
  },
}
`;

exports[`models/socket should export the events 1`] = `
Object {
  "BATTERY_LEVEL": Object {
    "description": "The current battery level of the device",
    "friendly": "Battery level",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "level": Object {
          "maxiumum": 100,
          "minimum": 0,
          "type": "double",
        },
      },
      "required": Array [
        "level",
      ],
      "type": "object",
    },
    "type": [Function],
  },
  "CONTACT": Object {
    "description": "Whether the contact sensor is closed or open",
    "friendly": "Contact",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "contact": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "contact",
      ],
      "type": "object",
    },
    "type": [Function],
  },
  "HUMIDITY": Object {
    "description": "The current humidity level",
    "friendly": "Humidity",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "level": Object {
          "maxiumum": 100,
          "minimum": 0,
          "type": "double",
        },
      },
      "required": Array [
        "level",
      ],
      "type": "object",
    },
    "type": [Function],
  },
  "LIGHT": Object {
    "description": "The current light level in lux",
    "friendly": "Light level",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "level": Object {
          "maxiumum": 30000,
          "minimum": 0,
          "type": "double",
        },
      },
      "required": Array [
        "level",
      ],
      "type": "object",
    },
    "type": [Function],
  },
  "MOTION": Object {
    "description": "Represents if motion is detected or not",
    "friendly": "Motion",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "detected": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "detected",
      ],
      "type": "object",
    },
    "type": [Function],
  },
  "NAME": Object {
    "description": "The current name of the device",
    "friendly": "Name",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "name": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
      "type": "object",
    },
    "type": [Function],
  },
  "TAMPER": Object {
    "description": "If the device is being tampered with or not",
    "friendly": "Tamper",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "detected": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "detected",
      ],
      "type": "object",
    },
    "type": [Function],
  },
  "TEMPERATURE": Object {
    "description": "The current temperature in degrees celcius",
    "friendly": "Temperature",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "level": Object {
          "maxiumum": 100,
          "minimum": -50,
          "type": "double",
        },
      },
      "required": Array [
        "level",
      ],
      "type": "object",
    },
    "type": [Function],
  },
  "UV": Object {
    "description": "The current UV level",
    "friendly": "UV level",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "level": Object {
          "maxiumum": 15,
          "minimum": 0,
          "type": "double",
        },
      },
      "required": Array [
        "level",
      ],
      "type": "object",
    },
    "type": [Function],
  },
  "VIBRATION": Object {
    "description": "The current vibration level",
    "friendly": "Vibration level",
    "responseSchema": Object {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "properties": Object {
        "level": Object {
          "maxiumum": 100,
          "minimum": 0,
          "type": "double",
        },
      },
      "required": Array [
        "level",
      ],
      "type": "object",
    },
    "type": [Function],
  },
}
`;
